pages/_app.tsx

```diff
@@ -4,19 +4,24 @@ import { AppProps } from 'next/app'
 import Head from 'next/head'
 import BeproService from '../services/bepro';
 import MainNav from '../components/main-nav';
-import React from 'react';
+import React, { useState } from 'react';
+import Loading from '../components/loading';
 
 export default function App({ Component, pageProps }: AppProps) {
+const [loading, setLoading] = useState<boolean>(false);
 
   const init = async () => {
     await BeproService.init();
   }
   init();
-  return <>
+
+  return ( 
+  <>
     <Head>
     </Head>
     <MainNav></MainNav>
-    <Component {...pageProps} />
-
+    <Component loading={setLoading} {...pageProps} />
```

Comment:
It'd be cleaner if `Loading` component exported a context of its own and then we could simply use it on all the application without having to connect `Component` to `Loading` since `Loading` would be using its own context that would be used by every other component that needs to show the loading

Reply:
Done